core-bundle/src/Controller/AbstractCustomBackendController.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller;
+
+use Contao\Ajax;
+use Contao\BackendMain;
+use Contao\BackendTemplate;
+use Contao\CoreBundle\Util\PackageUtil;
+use Contao\Environment;
+use Contao\Input;
+use Symfony\Component\HttpFoundation\Response;
+
+abstract class AbstractCustomBackendController extends AbstractController
+{
+    /**
+     * Render a Twig template with additional context for `@Contao/be_main`.
+     */
+    protected function render(string $view, array $parameters = [], Response $response = null): Response
+    {
+        $backendContext = (new class() extends BackendMain {
```

Comment:
What I don't like is that this PR builds on the `BackendMain` class 🙈 Or isn't this an issue because people extending `AbstractCustomBackendController` don't know about the legacy?

Reply:
Yes, it should be decoupled/implementation detail. But this is only a draft for now. I'm pretty sure there's more that can be done. 🙃